libs/api/books/src/lib/reading-list.service.ts

```diff
@@ -28,4 +28,18 @@ export class ReadingListService {
       return list.filter(x => x.bookId !== id);
     });
   }
+
+  async markBookAsFinished(id: string): Promise<ReadingListItem> {
+    let response: ReadingListItem = null;
+    this.storage.update(list => {
+      const item = list.find(x => x.bookId === id);
```

Comment:
A more functional way of doing this would be to use the `map` function.

Reply:
Refactored code to use `map` function